### PR TITLE
Explicit instruction sequencing

### PR DESCRIPTION
**Pull request by ehildenb**
This PR does a couple things:

-   Removes the rules for sequencing instructions in favor of explicit function calls which do it instead.
-   Updates the various proof tests to make the calls where appropriate.
-   Removes the `tests/bad-proofs`, since it just has one trivial proof there and we aren't really using it.
-   Adds `run ( _ ) => done ( _ )` to the functions proof tests, because it turns out that having 0-step proofs actually _can_ make the Haskell backend silently succeed a nonsense proof.

### CODE REVIEW

**Comment by ehildenb** [comment]
Hmmmm, actually @hjorthjort using `priority(25)` breaks the rule merger. Don't merge yet.

**Comment by ehildenb** [comment]
Ok @hjorthjort confirmed that the new version works with the rule-merger, this is good for review now.

**Comment by ehildenb** [comment]
On `master`:

```
* 8794dd4 - Fix typo (`eq` instead of `ne`) (#359)                                           - Jonas Schievink (upstream/master)
| testset: run - Fri Jun 12 23:00:45 UTC 2020
| 0  0.20s    0.01s     0.18s     3216KB     rm    -rf                         .build        deps
| 0  1.01s    0.60s     0.32s     32292KB    git   submodule                   update        --init        --recursive
| 0  219.77s  1661.32s  2258.64s  2883472KB  make  deps                        RELEASE=true
| 0  46.20s   504.35s   13.34s    3276060KB  make  build                       -j8           RELEASE=true
| 0  30.05s   755.58s   21.87s    892332KB   make  test-simple                 -j8
| 0  131.78s  4100.81s  57.04s    7482632KB  make  test-conformance-supported  -j8
| 0  133.33s  1434.69s  39.11s    3736292KB  make  test-prove                  -j8
```

This commit:

```
* 1bc6f7e - wasm, test: avoid use of priority(25)                                            - Everett Hildenbrandt (HEAD, upstream/sequence-instructions, testing, sequence-instructions)
| testset: run - Fri Jun 12 23:09:39 UTC 2020
| 0  0.32s    0.00s     0.30s     3220KB     rm    -rf                         .build        deps
| 0  0.90s    0.66s     0.26s     32220KB    git   submodule                   update        --init        --recursive
| 0  220.89s  1679.92s  2253.93s  2837124KB  make  deps                        RELEASE=true
| 0  43.29s   471.56s   13.13s    3194964KB  make  build                       -j8           RELEASE=true
| 0  30.61s   756.14s   22.12s    897584KB   make  test-simple                 -j8
| 0  122.41s  3859.68s  56.71s    8077364KB  make  test-conformance-supported  -j8
| 0  114.30s  1107.28s  31.78s    4031720KB  make  test-prove                  -j8
```

So perhaps marginally faster, but only a little bit if so, and completely possible it's just noise.

File: tests/proofs/locals-spec.k [hunk 1]
```diff
@@ -3,7 +3,7 @@ requires "kwasm-lemmas.md"
 module LOCALS-SPEC
     imports KWASM-LEMMAS
 
-    rule <k> (local.get X:Int) (local.set X:Int) => . ... </k>
+    rule <k> sequenceInstrs((local.get X:Int) (local.set X:Int)) => . ... </k>
```
Comment (hjorthjort) [inline tests/proofs/locals-spec.k:6]: I'd prefer to just sequence them directly with `~>` here, but up to you.

File: tests/proofs/simple-arithmetic-spec.k [hunk 2]
```diff
@@ -11,12 +11,12 @@ module SIMPLE-ARITHMETIC-SPEC
          <valstack> S:ValStack => < ITYPE > (X +Int #pow(ITYPE)) : S </valstack>
       requires (#minSigned(ITYPE) <=Int X) andBool (X <Int 0)
 
-    rule <k> (ITYPE:IValType . const X:Int) (ITYPE . const Y:Int) => . ... </k>
+    rule <k> sequenceInstrs((ITYPE:IValType . const X:Int) (ITYPE . const Y:Int)) => . ... </k>
```
Comment (hjorthjort) [inline tests/proofs/simple-arithmetic-spec.k:14]: Same here.

File: tests/proofs/simple-arithmetic-spec.k [hunk 3]
```diff
@@ -11,12 +11,12 @@ module SIMPLE-ARITHMETIC-SPEC
...
          <valstack> S:ValStack => < ITYPE > Y : < ITYPE > X : S </valstack>
       requires #inUnsignedRange(ITYPE, X)
        andBool #inUnsignedRange(ITYPE, Y)
 
-    rule <k> (ITYPE:IValType . const X:Int) (ITYPE . const Y:Int) (ITYPE . add) => . ... </k>
+    rule <k> sequenceInstrs((ITYPE:IValType . const X:Int) (ITYPE . const Y:Int) (ITYPE . add)) => . ... </k>
```
Comment (hjorthjort) [inline tests/proofs/simple-arithmetic-spec.k:19]: Same here

File: tests/proofs/locals-spec.k (hunk 1 above)
Comment (ehildenb) [reply tests/proofs/locals-spec.k]: Done

File: tests/proofs/simple-arithmetic-spec.k (hunk 2 above)
Comment (ehildenb) [reply tests/proofs/simple-arithmetic-spec.k]: Done

File: tests/proofs/simple-arithmetic-spec.k (hunk 3 above)
Comment (ehildenb) [reply tests/proofs/simple-arithmetic-spec.k]: Done

**Pull request merged**